core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/CongestionControl.scala

```diff
@@ -0,0 +1,64 @@
+package edu.uci.ics.amber.engine.architecture.messaginglayer
+
+import edu.uci.ics.amber.engine.architecture.messaginglayer.CongestionControl.{
+  maxWindowSize,
+  minWindowSize,
+  timeGap
+}
+import edu.uci.ics.amber.engine.architecture.messaginglayer.NetworkSenderActor.NetworkMessage
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+object CongestionControl {
+  final val maxWindowSize = 64
+  final val minWindowSize = 2
+  final val expendThreshold = 4
+  final val sendingTimeout: FiniteDuration = 30.seconds
+  final val timeGap = 3000 // 3s
+}
+
+class CongestionControl {
+
+  var ssThreshold = 16
+  var windowSize = minWindowSize
+  var sentTimeStamp = 0L
+  val toBeSent = new mutable.Queue[NetworkMessage]
+  val messageBuffer = new ArrayBuffer[NetworkMessage]()
+  private val inTransit = new mutable.LongMap[NetworkMessage]
+
+  def canBeSent(data: NetworkMessage): Boolean = {
```

Comment:
canSend() is shorter :)

Reply:
Actually this function doesn't just do canSend(). If the return is false, it also enqueues. So name should reflect that. 

Also, I had another question: Assume there is already data in toBeSent queue, when this data message comes. Then, why do we allow this message to jump the queue? There is a high probability that this message's seq num is after the messages in the queue. So, it will anyways be stashed in the receiving actor

Reply:
When receiving an ack, the messages in the queue will always be sent first. Then no message will jump the queue.